backend/WebApplication1/Services/FaceAPIService.cs

```diff
@@ -321,13 +258,14 @@ public async Task<List<FaceIdentifyResponse>> IdentifyAsync(string faceId, strin
                     List<FaceIdentifyResponse> result = JsonConvert.DeserializeObject<List<FaceIdentifyResponse>>(responseBody);
                     return result;
                 }
-                else
-                {
-                    string errorText = await response.Content.ReadAsStringAsync();
-                    ErrorResponse errorResponse = JsonConvert.DeserializeObject<ErrorResponse>(errorText);
-                    throw new HttpException(errorResponse.Code, errorResponse.Message);
-                }
+                throw CreateHttpException(await response.Content.ReadAsStringAsync());
             }
         }
+
+        private HttpException CreateHttpException(string errorText)
```

Comment:
a good idea to use factory method, but it shouldn't be in the service class, that's for sure. don't forget about single responsibility. I'd say just make it a static HttpException method